volatility3/framework/plugins/mac/dmesg.py

```diff
@@ -0,0 +1,78 @@
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+from volatility3.framework import interfaces, renderers
+from volatility3.framework.configuration import requirements
+from volatility3.framework.objects import utility
+
+vollog = logging.getLogger(__name__)
+
+
+class Dmesg(interfaces.plugins.PluginInterface):
+    """Prints the kernel log buffer."""
+
+    _required_framework_version = (2, 0, 0)
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Kernel module for the OS",
+                architectures=["Intel32", "Intel64"],
+            ),
+        ]
+
+    @classmethod
+    def get_kernel_log_buffer(
+        cls, context: interfaces.context.ContextInterface, kernel_module_name: str
+    ):
+        """
+        Online documentation :
+            - https://github.com/apple-open-source/macos/blob/master/xnu/bsd/sys/msgbuf.h
+            - https://github.com/apple-open-source/macos/blob/ea4cd5a06831aca49e33df829d2976d6de5316ec/xnu/bsd/kern/subr_log.c#L751
+        Volatility 2 plugin :
+            - https://github.com/volatilityfoundation/volatility/blob/master/volatility/plugins/mac/dmesg.py
+        """
+
+        kernel = context.modules[kernel_module_name]
+        if not kernel.has_symbol("msgbufp"):
+            raise TypeError(
```

Comment:
Hiya, you might want to raise an `exceptions.SymbolError` here, in case there's some way we could resolve the problem in the future, but I approve of throwing exceptions when something goes wrong, good work!  5:)
https://github.com/volatilityfoundation/volatility3/blob/887747e3ceb1d40af936e8a91d65e05874f4c06a/volatility3/framework/exceptions.py#L30